### PR TITLE
Basic Specs for all actions in VolunteersController

### PR DESCRIPTION
**Pull request by TimothyClayton**
1. tests template rendering and redirects
2. tests admin/non-admin allowed/disallowed for admin_only actions

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -26,10 +26,13 @@ group :development do
   gem 'rails-erd'
   gem 'binding_of_caller'
 end
+
 group :development, :test do
   gem 'pry-rails'
   gem 'pry-rescue'
   gem 'pry-remote'
+  gem 'byebug'
+  gem 'awesome_print'
```
Comment (bkuhlmann) [inline Gemfile]: +:star: Thanks!

File: Gemfile [hunk 2]
```diff
@@ -26,10 +26,13 @@ group :development do
   gem 'rails-erd'
   gem 'binding_of_caller'
 end
+
 group :development, :test do
   gem 'pry-rails'
   gem 'pry-rescue'
   gem 'pry-remote'
+  gem 'byebug'
```
Comment (rylanb) [inline Gemfile]: This makes sense to me

File: .gitignore [hunk 3]
```diff
@@ -11,3 +11,4 @@ public/system
 log/*.log.*
 .idea
 .bundle/*
+.byebug_history
```
Comment (bkuhlmann) [inline .gitignore]: :bulb: If `pry-byebug` is used instead, I don't think you'd have to add this line to the ignore file. I double checked my other projects and didn't see that I had added a line this to my `.gitignore` files so I think this is correct.

File: Gemfile (hunk 2 above)
Comment (TimothyClayton) [inline Gemfile]: aha, yeah wfm

File: spec/controllers/volunteers_controller_spec.rb [hunk 4]
```diff
@@ -2,8 +2,320 @@
...
+      end
+    end
+
+    context 'admin user' do
+      before { sign_in admin }
+
+      it 'renders the knight template' do
+        # TODO: fails with undef ar or meth `admin` in controller.
+        # Should be `admin_only`? Which is not a route. Or, `admin_only`
+        # should be `admin`?
```
Comment (bkuhlmann) [inline spec/controllers/volunteers_controller_spec.rb]: I'm not sure, either. Maybe mark this as a skipped spec until we can dive deeper?

File: .byebug_history [hunk 5]
```diff
@@ -0,0 +1,107 @@
+continue
```
Comment (bkuhlmann) [inline .byebug_history]: Ah, ok, I guess you do need to ignore this file.

File: .gitignore [hunk 6]
```diff
@@ -11,4 +11,3 @@ public/system
 log/*.log.*
 .idea
 .bundle/*
-.byebug_history
```
Comment (bkuhlmann) [inline .gitignore]: Sorry, I guess you should leave this in.

File: Gemfile [hunk 7]
```diff
@@ -31,7 +31,7 @@ group :development, :test do
   gem 'pry-rails'
   gem 'pry-rescue'
   gem 'pry-remote'
-  gem 'byebug'
+  gem 'pry-byebug'
```
Comment (bkuhlmann) [inline Gemfile]: ⭐️  Thanks.

File: .byebug_history [hunk 8]
```diff
@@ -1,107 +0,0 @@
-continue
```
Comment (bkuhlmann) [inline .byebug_history]: You can ignore my previous comment...just saw this commit. If you want, you might want to interactive rebase your commits and make this comment be part of the commit where this was accidentally checked in. This way the Git history will be cleaned up to make it look like this was never checked in. Example (i.e. `git rebase --interactive "@~4`):

```
pick 96773f3 replaces byebug with pry-byebug
fixup 2579acd removes byebug history

# Rebase a3a10ab..2579acd onto a3a10ab (4 commands)
#
# Commands:
# p, pick = use commit
# r, reword = use commit, but edit the commit message
# e, edit = use commit, but stop for amending
# s, squash = use commit, but meld into previous commit
# f, fixup = like "squash", but discard this commit's log message
# x, exec = run command (the rest of the line) using shell
# d, drop = remove commit
```

**Comment by bkuhlmann** [comment]
:information_source: Done reviewing. These changes seem reasonable to me. Only had some minor suggestions for Git Log cleanup but otherwise this seems reasonable to merge. Will defer to @rylanb for final approval.

File: .byebug_history (hunk 8 above)
Comment (TimothyClayton) [inline .byebug_history]: Got it. When these PR get merged I'm going to stop submitting from my fork 😛

File: spec/controllers/volunteers_controller_spec.rb (hunk 4 above)
Comment (TimothyClayton) [inline spec/controllers/volunteers_controller_spec.rb]: Agreed. I will make it a pending spec once it is merged and I pull down the base repo.

**Pull request merged**